### PR TITLE
Clarify the error message for identityName vs identity.name conflicts

### PR DESCRIPTION
**Pull request by patrick-codaio**

### CODE REVIEW

File: test/api_test.ts [hunk 1]
```diff
@@ -184,7 +184,7 @@ describe('API test', () => {
               },
             },
           }),
-        "Sync table defines an identityName that conflicts with its schema's identity.name",
+        "Sync table SomeSync's schema's identity has a name (ConflictingIdentity) which must match the identityName of the table (MyIdentityName).",
```
Comment (ekoleda-codaio) [inline test/api_test.ts:187]: Better, but "SomeSync's schema's" is a little hard to parse. Riffing:

> Identity name mismatch for sync table SomeSync. Either remove the schema's identity.name (ConflictingIdentity) or ensure it matches the table's identityName (MyIdentityName).

CC @alicia-codaio who may have wordsmithing suggestions.

Comment (patrick-codaio) [reply test/api_test.ts:187]: That's fantastic wording, done!

**Review by ekoleda-codaio — APPROVED** [review]
